### PR TITLE
Fix MetricsUploader

### PR DESCRIPTION
**Pull request by IrinaShkviro**
SetupConnection has one argument - string that will be used as suffix in
the metrics_uploader_client log file.

Removed FreeLibrary call that leads to crash.

### CODE REVIEW

File: src/MetricsUploader/MetricsUploaderWindows.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ namespace orbit_metrics_uploader {
 constexpr const char* kSendLogEventFunctionName = "SendOrbitLogEvent";
 constexpr const char* kSetupConnectionFunctionName = "SetupConnection";
 constexpr const char* kShutdownConnectionFunctionName = "ShutdownConnection";
+constexpr const char* kClientLogFileSuffix = "Orbit";
```
Comment (beckerhe) [inline src/MetricsUploader/MetricsUploaderWindows.cpp:19]: Can you add some more details to the commit message?
Where is that logfile stored? What is the actual filename? And where does the suffix come into play?

File: src/MetricsUploader/MetricsUploaderWindows.cpp [hunk 2]
```diff
@@ -65,9 +58,6 @@ MetricsUploader::~MetricsUploader() {
       ERROR("Error while closing connection: %s", GetErrorMessage(result));
     }
   }
-  if (nullptr != metrics_uploader_client_dll_) {
-    FreeLibrary(metrics_uploader_client_dll_);
-  }
```
Comment (beckerhe) [inline src/MetricsUploader/MetricsUploaderWindows.cpp:69]: The code should stay here. FreeLibrary should be commented and a comment should be added explaining why we can't call it.

File: src/MetricsUploader/MetricsUploaderWindows.cpp [hunk 3]
```diff
@@ -20,21 +20,17 @@ constexpr const char* kClientLogFileSuffix = "Orbit";
...
                                  Result (*send_log_event_addr)(const uint8_t*, int),
-                                 Result (*shutdown_connection_addr)(),
-                                 HMODULE metrics_uploader_client_dll)
+                                 Result (*shutdown_connection_addr)())
     : client_name_(std::move(client_name)),
       session_uuid_(std::move(session_uuid)),
       send_log_event_addr_(send_log_event_addr),
-      shutdown_connection_addr_(shutdown_connection_addr),
-      metrics_uploader_client_dll_(metrics_uploader_client_dll) {}
+      shutdown_connection_addr_(shutdown_connection_addr) {}
```
Comment (beckerhe) [inline src/MetricsUploader/MetricsUploaderWindows.cpp:27]: I believe the HMODULE should stay here, even if we currently don't need it.

This class acts as the owner of that handle and we shouldn't throw it away.

**Review by beckerhe — COMMENTED** [review]
Thank you Irina. I left some comments. :-)

File: src/MetricsUploader/MetricsUploaderWindows.cpp (hunk 3 above)
Comment (IrinaShkviro) [reply src/MetricsUploader/MetricsUploaderWindows.cpp]: ok, thanks. I've left the HMODULE as a class member

File: src/MetricsUploader/MetricsUploaderWindows.cpp (hunk 2 above)
Comment (IrinaShkviro) [reply src/MetricsUploader/MetricsUploaderWindows.cpp:69]: Updated the code: commented FreeLibrary calls and add explanation in the destructor as well

File: src/MetricsUploader/MetricsUploaderWindows.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply src/MetricsUploader/MetricsUploaderWindows.cpp:19]: Updated the commit message

**Comment by IrinaShkviro** [comment]
> Thank you Irina. I left some comments. :-)

Thank you for taking a look! I've updated the code

**Review by beckerhe — APPROVED** [review]
I had another look. LGTM!

**Review by antonrohr — APPROVED** [review]
LGTM!!!

**Pull request merged**